### PR TITLE
dffml: source: file: Added Gzip support

### PR DESCRIPTION
**Pull request by yashlamba**
Added condition for opening and closing gzip files

### CODE REVIEW

**Comment by yashlamba** [comment]
@pdxjohnny So what I did was I used the iris dataset for checking the gzip support. I compressed iris_training.csv to iris_training.csv.gz. 
I encountered a few errors fixed in commit 6da2fd9 and e97eea3. After building it again, I ran the following command:
```
dffml train \
  -model dnn \
  -sources csv=/home/nucleon/iris_training.csv.gz
  -classifications 0 1 2 \
  -features \
    def:SepalLength:float:1 \
    def:SepalWidth:float:1 \
    def:PetalLength:float:1 \
    def:PetalWidth:float:1 \
  -num_epochs 3000 \
  -steps 20000 \
  -log debug
```
And got the following result:
Screenshot from 2019-03-22 17-05-38

Does this look good?

**Comment by pdxjohnny** [comment]
Awesome!! Nice work! That was a good way to test it. Let's include a testcase within the codebase as well.

See [tests/source/test_csv.py](https://github.com/intel/dffml/blob/master/tests/source/test_csv.py) for an example.

This is along the lines of what you'll want, but not exactly (i haven't tried this I'm just trying to illustrate the basic structure or what this will look like.

```python
        # Repo definitions ...

        with tempfile.NamedTemporaryFile() as testfile, \
                tempfile.NamedTemporaryFile(suffix='.gz') as testfile_gz:
            source = JSONSource(testfile.name)
            # Open, update, and close
            async with source as sctx:
                await sctx.update(full_repo)
                await sctx.update(empty_repo)
            with open(testfile.name, 'rb') as testfile_fd:
                with gzip.open(testfile_gz.name, 'wb') as testfile_gz_fd:
                    shutil.copyfileobj(testfile_fd, testfile_gz_fd)
            source = JSONSource(testfile_gz.name)
            # Open and confirm we saved and loaded correctly
            async with source as sctx:
                with self.subTest(src_url=full_src_url):
                    repo = await sctx.repo(full_src_url)
                    self.assertEqual(repo.data.prediction.classification,
                                     "feedface")
                    self.assertEqual(repo.data.prediction.confidence, 0.42)
                with self.subTest(src_url=empty_src_url):
                    repo = await sctx.repo(empty_src_url)
                    self.assertFalse(repo.data.prediction.classification)
                    self.assertFalse(repo.data.prediction.confidence)
```

**Comment by yashlamba** [comment]
I might be wrong but just want to clear my head over it, why don't we just modify test_file.py with another case? I have just started understanding unittest and that was what I was anticipating. 
I got what you said above too and will start with that now.

**Comment by pdxjohnny** [comment]
Yes sure, modify test_file, that sounds good.

File: tests/source/test_file.py [hunk 1]
```diff
@@ -76,6 +84,13 @@ def test_repr(self):
             await source.close()
             m_open.assert_called_once_with('testfile', 'w')
 
+    async def test_close_gz(self):
+        source = FakeFileSource('testfile.gz')
+        m_open = mock_open()
+        with patch('builtins.open', m_open):
+            await source.close()
+            m_open.assert_called_once_with('testfile.gz', 'wb')
```
Comment (pdxjohnny) [inline tests/source/test_file.py]: This is a much better idea!!

Comment (yashlamba) [reply tests/source/test_file.py]: Is there anything else to be done for gzip support? Or shall I start working on the next?

**Comment by yashlamba** [comment]
Is there anything else to be done for gzip support? Or shall I start working on the next?

I'll remove WIP from it then.

**Comment by pdxjohnny** [comment]
Nope this looks great! However, can you add this under the `Changes` section of the CHANGELOG?

Thanks!!!

**Comment by yashlamba** [comment]
Done! 2cb2ed6

**Comment by pdxjohnny** [comment]
Well done @yashlamba ! This looks great. Merged.

**Pull request merged**